File: app/src/main/java/com/example/graduationprojectandroid/network/endpoints/APIs/TasksAPI.java
package com.example.graduationprojectandroid.network.endpoints.APIs;

import com.google.gson.JsonObject;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface TasksAPI {
    @POST("/set_task_state")
    Call<String> setTaskState(
            @Query("task_id") int taskId,
            @Query("task_state") int taskState,
            @Query("token") String token
    );

    @POST("/create_task")
    Call<String> createTask(
            @Body JsonObject task,
            @Query("token") String token
    );

    @POST("/edit_task")
    Call<String> editTask(
            @Body JsonObject task,
            @Query("token") String token
    );

    @GET("/get_tasks")
    Call<String> getTasks(
            @Query("login_from") String loginFrom,
            @Query("login_to") String loginTo,
            @Query("token") String token
    );
}

File: app/src/main/java/com/example/graduationprojectandroid/network/endpoints/APIs/TasksAPI.kt
package com.example.graduationprojectandroid.network.endpoints.APIs

import com.example.graduationprojectandroid.data.Items.Task
import com.example.graduationprojectandroid.network.endpoints.ServerAnswer
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.Query

interface TasksAPI {
    @POST("/set_task_state")
    fun setTaskState(
        @Query("task_id") taskId: Int,
        @Query("task_state") taskState: Int,
        @Query("token") token: String?
    ): Call<ServerAnswer>

    @POST("/create_task")
    fun createTask(
        @Body task: Task?,
        @Query("token") token: String?
    ): Call<ServerAnswer>

    @POST("/edit_task")
    fun editTask(
        @Body task: Task?,
        @Query("token") token: String?
    ): Call<ServerAnswer>

    @GET("/get_tasks")
    fun getTasks(
        @Query("login_from") loginFrom: String?,
        @Query("login_to") loginTo: String?,
        @Query("token") token: String?
    ): Call<ServerAnswer>
}